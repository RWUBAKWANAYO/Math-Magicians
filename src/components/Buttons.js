import React from 'react';
import PropTypes from 'prop-types';

const Buttons = (props) => {
  const { handleClick } = props;
  return (
    <div className="calculator-wrapper">
      <div className="calculator-numbers-cont">
        <div className="calculator-numbers" id="special-character">
          <button type="button" onClick={() => handleClick('AC')}>AC</button>
          <button type="button" onClick={() => handleClick('+/-')}>+/-</button>
          <button type="button" onClick={() => handleClick('%')}>%</button>
        </div>
        <div className="calculator-numbers">
          <button type="button" onClick={() => handleClick('7')}>7</button>
          <button type="button" onClick={() => handleClick('8')}>8</button>
          <button type="button" onClick={() => handleClick('9')}>9</button>
        </div>
        <div className="calculator-numbers">
          <button type="button" onClick={() => handleClick('4')}>4</button>
          <button type="button" onClick={() => handleClick('5')}>5</button>
          <button type="button" onClick={() => handleClick('6')}>6</button>
        </div>
        <div className="calculator-numbers">
          <button type="button" onClick={() => handleClick('1')}>1</button>
          <button type="button" onClick={() => handleClick('2')}>2</button>
          <button type="button" onClick={() => handleClick('3')}>3</button>
        </div>
        <div className="calculator-numbers">
          <button type="button" id="zero-button" onClick={() => handleClick('0')}>0</button>
          <button type="button" onClick={() => handleClick('.')}>.</button>
        </div>
      </div>
      <div className="calculator-symbol-cont">
        <button type="button" onClick={() => handleClick('รท')}>รท</button>
        <button type="button" onClick={() => handleClick('x')}>ร</button>
        <button type="button" onClick={() => handleClick('-')}>-</button>
        <button type="button" onClick={() => handleClick('+')}>+</button>
        <button type="button" onClick={() => handleClick('=')}>=</button>
      </div>
    </div>
  );
};

Buttons.propTypes = { handleClick: PropTypes.func };
Buttons.defaultProps = { handleClick: '' };

export default Buttons;
