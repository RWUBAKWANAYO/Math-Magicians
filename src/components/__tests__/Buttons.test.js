import renderer from 'react-test-renderer';
import Buttons from '../Buttons';

it('Button component render', () => {
  const tree = renderer.create(<Buttons handleClick={(a = '1') => a} />).toJSON();
  expect(tree).toMatchSnapshot();
});
