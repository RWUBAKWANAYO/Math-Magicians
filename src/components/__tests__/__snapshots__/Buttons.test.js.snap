// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component render 1`] = `
<div
  className="calculator-wrapper"
>
  <div
    className="calculator-numbers-cont"
  >
    <div
      className="calculator-numbers"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div
      className="calculator-numbers"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="calculator-numbers"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="calculator-numbers"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="calculator-numbers"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
  </div>
  <div
    className="calculator-symbol-cont"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      ร
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
